usl_models/usl_models/flood_ml/model.py

```diff
@@ -68,57 +67,11 @@ def _build_model(self) -> keras.Model:
         )
         return model
 
-    def _validate_and_preprocess_data(
-        self,
-        data: FloodModelData,
-        training=True,
-    ) -> FloodModelData:
-        """Validates model data and does all necessary preprocessing.
-
-        Args:
-            data: A FloodModelData object. Labels are required for training.
-            training: Whether data is used for model training. If True, labels
-                will be validated.
-
-        Returns:
-            A processed FloodModelData object.
-        """
-        if training:
-            # Labels are required for training.
-            if data.labels is None:
-                raise ValueError("Labels must be provided during model training.")
-
-            # Labels must match the storm duration.
-            expected_label_shape = list(data.labels.shape)
-            expected_label_shape[1] = data.storm_duration
-            assert data.labels.shape[1] == data.storm_duration, (
-                "Provided labels are inconsistent with storm duration. "
-                f"Labels are expected to have shape {expected_label_shape}. "
-                f"Actual shape: {data.labels.shape}."
-            )
-
-        # Check whether the temporal data is already windowed. If it is, checks
-        # the expected shape. Otherwise, create the window view.
-        if tf.rank(data.temporal) == 3:  # windowed: (B, T_max, m)
-            assert data.temporal.shape[-1] == self._model_params["m_rainfall"], (
-                "Mismatch between the temporal data window size "
-                f"({data.temporal.shape[-1]}) and the expected window size "
-                f"(m = {self._model_params['m_rainfall']})."
-            )
-        else:
-            full_temp_input = data_utils.temporal_window_view(
-                data.temporal, self._model_params["m_rainfall"]
-            )
-            data = dataclasses.replace(data, temporal=full_temp_input)
-
-        # We assume that, if provided, this input is a *single* flood map.
-        st_input = data.spatiotemporal
-        if st_input is None:
-            st_shape = data.geospatial.shape[:3] + [1]
-            st_input = tf.zeros(st_shape)
-
-        data = dataclasses.replace(data, spatiotemporal=st_input)
-        return data
+    def _load_weights_from_artifact(self, artifact_uri: str):
+        """Loads model weights from a GCS artifact URI."""
+        prediction_utils.download_model_artifacts(artifact_uri)
+        loaded_model = keras.models.load_model("model")
```

Comment:
We can do 
```python
 keras.models.load_model('gs://climateiq-vertexai/aiplatform...')
```
to grab the model without using `download_model_artifacts`
I'd prefer not to use `download_model_artifacts` because it writes things out to the local disk.

Reply:
Gotcha, I didn't realize this was possible. Good to know!